ssl/record/ssl3_record.c

```diff
@@ -711,16 +711,20 @@ int ssl3_get_record(SSL *s)
             goto end;
         }
 
+        /*
+         * Usually thisrr->length is accurate, just not when KTLS receives
+         * adjacent maximum-size packets.
+         * Just assume the SSL3_RT_MAX_PLAIN_LENGTH is checked in the kernel.
+         */
         if (thisrr->length > SSL3_RT_MAX_PLAIN_LENGTH
-            && !BIO_get_ktls_recv(s->rbio)) {
+                && (!BIO_get_ktls_recv(s->rbio) || is_ktls_left)) {
             SSLfatal(s, SSL_AD_RECORD_OVERFLOW, SSL_R_DATA_LENGTH_TOO_LONG);
             goto end;
         }
 
         /* If received packet overflows current Max Fragment Length setting */
         if (s->session != NULL && USE_MAX_FRAGMENT_LENGTH_EXT(s->session)
-                && thisrr->length > GET_MAX_FRAGMENT_LENGTH(s->session)
-                && !BIO_get_ktls_recv(s->rbio)) {
+                && thisrr->length > GET_MAX_FRAGMENT_LENGTH(s->session)) {
```

Comment:
That caused my attention initially as this check is required by RFC 6066

Reply:
max_fragment_length and KTLS are mutually exclusive. You will never (with the currently KTLS support) have max_fragment_length and KTLS being used at the same time.

Reply:
Ah okay, that explains why this never happens with KTLS.

Reply:
Should I add a comment for this?

Reply:
Yes - good idea